crates/full-node/sov-sequencer/src/preferred/side_effects.rs

```diff
@@ -60,15 +60,15 @@ where
         batch: PreferredSequencerReadBatch,
         info_to_store: BatchToStore,
     ) -> anyhow::Result<()> {
-        self.db.terminate_batch(info_to_store).await?;
-        self.update_api_state(checkpoint);
-
         // Publish the batch.
         self.blob_sender
             .add_txs(batch.blob_id, batch.tx_hashes.clone())
             .await;
         self.blob_sender.publish_batch(batch).await?;
 
+        self.db.terminate_batch(info_to_store).await?;
```

Comment:
Now we do `self.blob_sender.publish_batch(batch).await?` before `terminate_batch` so the `blob_sender` keeps the batch in the internal db even if the sequencer crashes.